ext/serializer.c

```diff
@@ -319,11 +312,13 @@ static int dd_fatal_error_to_meta(zend_array *meta, dd_error_info error) {
     return error.type && error.msg ? SUCCESS : FAILURE;
 }
 
-static zend_result dd_add_meta_array(void *context, ddtrace_string key, ddtrace_string value) {
-    zval *meta = context, tmp = ddtrace_zval_stringl(value.ptr, value.len);
+static zend_result dd_add_meta_array(ddog_SpanBytes *span, ddtrace_string key, ddtrace_string value) {
```

Comment:
We should probably get rid of this function and just call ddog_add_span_meta and such variants at the appropriate places, and also avoid passing this around as function pointer.
This function had more uses in the past, but now it can be eliminated, when we're refactoring this anyway now.